Make line clearing more robust, add a note in the CHANGELOG

LukeShu opened the pull request:
These 2 changes aren't related, but they're both small enough that making separate PRs for them felt silly.

LukeShu committed:
ui: Be more robust about clearing the line

Some terminals (like urxvt) will allow you to "backspace" (\x08) past the
beginning of the current line, in to the previous line.  This means that
each update to the status line moves farther and farther *up* the screen.

Instead, prefix each line with "\r" to move the cursor to the beginning of
the line, and suffix each line with the "\1B[K" escape sequence to clear
from the cursor to the end of the line.

While we're at it, add comments explaining each of the escape sequences.

v2:
 - Use "\0[K" instead of overwriting with whitespace.
 - Add comments.
 - Add a CHANGELOG entry.

LukeShu committed:
CHANGELOG: Add a note about upgrading to 2.x

majewsky commented on ui.go:
```diff
@@ -20,35 +20,44 @@ package main
+}
+
+func (ui *UI) setLine(line string) {
+	fmt.Printf("\r%-[1]*[2]s", ui.prevLen, line)
```
What's going on here? This doesn't parse in my head, and Github's syntax highlighting flags the `%` as an error (presumably an incomplete formatting directive).

LukeShu replied:
* `-` pad with spaces on the right rather than the left (left-justify the field)
* `*` get the width to whitespace-pad to from an argument, rather than from the format string itself
* `[1]` identifies which argument to get the width from
* `[2]` identifies which argument to get the string contents from

It could be written without the `[1]` and `[2]`, but they help me when reading it.

```go
fmt.Printf("\r%-*s", ui.prevLen, line)
```

LukeShu committed:
Switch from golangvend to Go modules

LukeShu commented:
I've
1. Rebased this on to master
2. Adjusted it to use escape codes to clear the rest of the line, rather than overwriting with spaces.
3. Switched it from golangvend to Go modules
